### PR TITLE
[build-script] Add plumbing for llbuild assertions

### PR DESCRIPTION
**Pull request by aciidgh**
This adds two flags to the build script to enable/disable assertions in
llbuild: --llbuild-assertions, --no-llbuild-assertions

The default value is taken from the global assertions flag.

### CODE REVIEW

**Comment by aciidb0mb3r** [comment]
Depends on https://github.com/apple/swift-llbuild/pull/232

File: utils/build-script-impl [hunk 1]
```diff
@@ -2354,7 +2354,7 @@ for host in "${ALL_HOSTS[@]}"; do
                     -DLIT_EXECUTABLE:PATH="${LLVM_SOURCE_DIR}/utils/lit/lit.py"
                     -DFILECHECK_EXECUTABLE:PATH="$(build_directory_bin ${LOCAL_HOST} llvm)/FileCheck"
                     -DCMAKE_BUILD_TYPE:STRING="${LLBUILD_BUILD_TYPE}"
-                    -DLLVM_ENABLE_ASSERTIONS:BOOL=$(true_false "${LLBUILD_ENABLE_ASSERTIONS}")
+                    -DLLBUILD_ENABLE_ASSERTIONS:BOOL=$(true_false "${LLBUILD_ENABLE_ASSERTIONS}")
```
Comment (aciidb0mb3r) [inline utils/build-script-impl]: note: `llbuild`'s CMake doesn't actually use `LLVM_ENABLE_ASSERTIONS`.

**Review by graydon — APPROVED** [review]
Yup, something like that! Thanks :)

**Review by ddunbar — APPROVED** [review]

**Review by ddunbar — APPROVED** [review]

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

**Comment by jrose-apple** [comment]
The pythonlint one totally could be…

@swift-ci Please Python lint

**Comment by aciidb0mb3r** [comment]
Wow, ok.
```
******************** TEST 'Swift(macosx-x86_64) :: Python/python-lint.swift' FAILED ********************
Script:
--
/usr/bin/python /Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/swift/utils/python_lint.py
--
Exit Code: 1

Command Output (stdout):
--
./utils/build_swift/tests/expected_options.py:372:80: E501 line too long (81 > 79 characters)

--

********************
```

**Comment by aciidb0mb3r** [comment]
@swift-ci Please Python lint

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

**Pull request merged**